independent-projects/bootstrap/maven-plugin/src/main/java/io/quarkus/maven/ExtensionDescriptorMojo.java

```diff
@@ -401,23 +392,21 @@ private void completeCodestartArtifact(ObjectMapper mapper, ObjectNode extObject
             }
             return;
         }
-        org.eclipse.aether.artifact.Artifact codestartArtifact = DependencyNodeUtils.toArtifact(mvalue.asText());
 
-        if (codestartArtifact.getVersion() == null || codestartArtifact.getVersion().isEmpty()) {
-            codestartArtifact = codestartArtifact.setVersion(project.getVersion());
-            codestartObject.put("artifact",
-                    codestartArtifact.getGroupId() + ":" + codestartArtifact.getArtifactId() + ":"
-                            + codestartArtifact.getClassifier() + ":" + codestartArtifact.getExtension() + ":"
-                            + codestartArtifact.getVersion());
```

Comment:
Do you mind adding a comment about what is expected here and what is not supported?

Reply:
@aloubyansky sure, in another PR?

"G:A" or anything supported by AppArtifactCoords.fromString() with `${project.version}` support.

Reply:
Doesn't look like it's worth creating a separate PR for a comment. Could you add it to this one?

Reply:
Ok everything's fine beside that?

Reply:
I just hate to wait for the CI for a comment

Reply:
Yes, looks good, thanks!